Add link to documentation in tsconfig.json template generated by tsc --init

borgfriend opened the pull request:
<!--
Thank you for submitting a pull request!

Please verify that:
* [ ] There is an associated issue in the `Backlog` milestone (**required**)
* [ ] Code is up-to-date with the `master` branch
* [ ] You've successfully run `gulp runtests` locally
* [ ] There are new or updated unit tests validating the change

Refer to CONTRIBUTING.MD for more details.
  https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md
-->

Fixes #34538

RyanCavanaugh commented:
A PR adding the link to the top is OK, but again, we do not want literally every compiler flag listed in the init tsconfig. For example, some of the flags listed here *do nothing*, are *explicitly discouraged*, or have *dangerous effects*.

DanielRosenwasser commented:
Reopening in case you want to just make the PR about providing the link to other options. In that case, I would also suggest we remove the comment descriptions because they tend to be a major pain when using `tsc --init`.

borgfriend commented:
I removed the Advanced Options again... however I would suggest to move various options then into better category names like "Deprecated", "Use with Caution" etc.

Me not being really familiar with the codebase just assumed these are all "Advanced options" that you should be able to use as an advanced user.

In addition I removed the comment descriptions as you suggested.

sandersn commented:
@borgfriend Can you update the description of this PR? I'm triaging our old PRs that we forgot about and I can't tell what this evolved into, but it seems different than the initial description.

borgfriend commented:
I think it is quite misleading to only expose a subset of flags available in tsconfig.json when generating it via tsc --init. 

The current flags are "handpicked" because - there are dangerous, deprecated flags that should not be exposed.

The issue evolved into:
- add link to documentation
- remove documentation strings 

Thus users would be encouraged to take a look at the documentation to learn more about available flags (also those that are not available in the 'handpicked selection')

sheetalkamat commented:
@borgfriend  can you please update this to resolve merge conflicts.. Thanks

orta commented:
There's an example of what we'd like it to look like in https://github.com/microsoft/TypeScript/issues/37361#issuecomment-598867169 too

borgfriend commented:
@sheetalkamat I fixed the merge conflicts, @orta adjusted the link to be like in the example

sheetalkamat commented on src/compiler/commandLineParser.ts:
```diff
@@ -150,6 +150,14 @@ namespace ts {
             category: Diagnostics.Command_line_Options,
             description: Diagnostics.Whether_to_keep_outdated_console_output_in_watch_mode_instead_of_clearing_the_screen,
         },
+        {
```
There is already incremental included in this list why is there another entry?

borgfriend replied:
guess that was automerged and I missed it - pushed a fix for it

sheetalkamat replied:
Can you please existing entry instead? Is there reason this needs to move ?

borgfriend replied:
better now?

sheetalkamat commented on tests/baselines/reference/tsConfig/Default initialized TSConfig/tsconfig.json:
```diff
@@ -1,66 +1,69 @@
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
     /* Basic Options */
-    // "incremental": true,                   /* Enable incremental compilation */
```
Example shows the comments as not removed https://github.com/microsoft/TypeScript/issues/37361#issuecomment-598867169 @RyanCavanaugh @DanielRosenwasser Whats the final call on that

DanielRosenwasser commented on src/compiler/commandLineParser.ts:
```diff
@@ -2108,10 +2108,9 @@ namespace ts {
 
         function isAllowedOption({ category, name }: CommandLineOption): boolean {
             // Skip options which do not have a category or have category `Command_line_Options`
-            // Exclude all possible `Advanced_Options` in tsconfig.json which were NOT defined in command line
```
Return the comment, remove the extra `;`

borgfriend replied:
done

sheetalkamat replied:
The comment is still gone. Can you please fix that so that its ready to merge

sheetalkamat commented on tests/baselines/reference/tsConfig/Default initialized TSConfig/tsconfig.json:
```diff
@@ -1,66 +1,68 @@
-    // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
-    // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
-    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
+    /* Basic Options */                       /* Enable incremental compilation */
```
The comments are off by a line ?

borgfriend replied:
This is rather odd, i just reverted my change of removing them.
I adjusted the code so an offset is not possible anymore

borgfriend commented:
I just noticed that for the flag "target" the description states the default is 'es3' - however it is set to 'es5'? 

Maybe the target and description should be updated to ES2017 (version supported by node 8, and all current Browsers without IE)?

Should I change the flag?

sheetalkamat commented:
@borgfriend default for target is es3 (that means if target is not specified it is es3). The comment is correct
